### DISCUSSION TITLE
Using different resize filters per channel

### DISCUSSION THREAD

**Discussion by frugbug** [ImageMagick/ImageMagick#4534] [Help]
I have some images which need to be downscaled. I need to resize the red and blue channels smoothly, but the green and alpha channels have to keep existing values (so I would use point/nearest filtering, as interpolation would result in new values being introduced). Here is what I've been doing:
`magick convert input.png -channel R,B -filter catrom -define filter:blur=3 -resize 64x64 output.png`
I'm pretty sure this is not the correct way to do it, but I'm unsure of another way which doesn't involve writing to temp files. This command actually does work, but produces inconsistent results. It works as I expected for 8 bit images (NOT 8 bit per channel). The blurry catrom filter is applied to the R and B channels and the G channel looks like it uses point filtering. However, on images with 8 bits per channel (24bit or 32bit for the whole image), the catrom filter applies to all channels.
Here is a zip file containing the input and output images:
[IM problem.zip](https://github.com/ImageMagick/ImageMagick/files/7638950/IM.problem.zip)

My goal is to have a command which downscales the red and blue channels using the catrom filter, but uses a point filter for green and alpha. I'm using IM 7.0.10-25 on windows

**Comment by snibgo** [comment]
Nothing in your command specifies point filtering.

"-resize" uses the same set of parameters for all channels.

Your images have no transparency, so there is no need for special processing to alpha.

If you want different filtering parameters for the different channels, I suggest separating the channels, doing a resize on each with whatever parameters you want, then combining them.

**Reply by fmw42** [reply]
Use clones or mpr: and parenthesis processing to avoid the temp files. See https://legacy.imagemagick.org/Usage/basics/#parenthesis

**Reply by frugbug** [reply]
> Use clones or mpr: and parenthesis processing to avoid the temp files. See https://legacy.imagemagick.org/Usage/basics/#parenthesis

@fmw42 Could you please give an example of how to do this?

**Reply by fmw42** [reply]
Input:

![test_24bit_s](https://user-images.githubusercontent.com/13458218/145242413-c9385de3-f0c5-4683-8570-e76695bb71d2.png)

Unix syntax:

```
magick test_24bit_s.png -separate \
\( -clone 0,2 -filter catrom -resize 64x64 \) \
\( -clone 1 -filter point -resize 64x64 \) \
-delete 0-2 +swap -combine \
PNG24:test_24bit_s_64.png

```

![test_24bit_s_64](https://user-images.githubusercontent.com/13458218/145242524-dd599587-0d51-4562-bff3-1ff49a96071c.png)

If on windows, remove the \ from the parentheses and change the end of line \ to ^

**Reply by frugbug** [reply]
Thanks very much. Although that didn't really work for me I tweaked a few parameters and got what I wanted. In the end I used this command:
 `magick input.png -separate ( -clone 0,2 -filter catrom -resize 64x64 ) ( -clone 1,3 -filter point -resize 64x64 ) -delete 0,1,2,3 -swap 1,2 -combine output.png`
